lib/src/components/NxTag/__tests__/NxTag.test.tsx

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React from 'react';
+import * as enzymeUtils from '../../../__testutils__/enzymeUtils';
+import { shallow } from 'enzyme';
+import { faPlusCircle, faTimesCircle } from '@fortawesome/free-solid-svg-icons';
+
+import NxTag, { NxSelectableTag } from '../NxTag';
+import NxFontAwesomeIcon from '../../NxFontAwesomeIcon/NxFontAwesomeIcon';
+
+describe('NxTag', function() {
+  const getShallowComponent = enzymeUtils.getShallowComponent(NxTag, { children: 'basic tag' });
+
+  it('renders NxTag with the `nx-tag` class along with any provided className', function() {
+    expect(getShallowComponent().find('.nx-tag')).toExist();
+  });
+
+  it('correctly assigns supplied class', function() {
+    const classComponent = getShallowComponent({ className: 'foo' });
+    expect(classComponent.find('.nx-tag')).toHaveClassName('foo');
+  });
+
+  it('correctly assigns supplied id', function() {
+    const idComponent = getShallowComponent({ id: 'test-id' });
+    expect(idComponent.find('.nx-tag')).toHaveProp('id', 'test-id');
+  });
+
+  it('sets the nx-tag--default class if no color prop is passed', function() {
+    expect(getShallowComponent().find('.nx-tag--default')).toExist();
+  });
+
+  it('sets the color class using the color if it is provided', function() {
+    const colorComponent = getShallowComponent({ color: 'orange' });
+    expect(colorComponent.find('.nx-tag--orange')).toExist();
+    expect(colorComponent.find('.nx-tag--default')).not.toExist();
+  });
+});
```

Comment:
Need a test that `NxTag` renders its children in an `.nx-tag__text`.  And another test that it renders the `selectedIcons` prop after the `.nx-tag__text`

Reply:
> Need a test that `NxTag` renders its children in an `.nx-tag__text`. And another test that it renders the `selectedIcons` prop after the `.nx-tag__text`

Test for children inside `.nx-tag__text`: https://github.com/sonatype/sonatype-react-shared-components/pull/290/commits/1f8beba3c765ff410e13ce62300f6ce946dd376f

Test that the text is the first child (i.e. it appears before the icon): https://github.com/sonatype/sonatype-react-shared-components/pull/290/commits/597f949381fd36bf9a6681f57984e7127cc12ddc

Reply:
> Test that the text is the first child (i.

That commit does not succeed in its purpose.  Observe that if you change the implementation to put the icon before the text the tests will still pass.  You need a test of `NxTag` (not `NxSelectableTag`) in which the `selectedIcons` prop is used and the `.children()` are inspectd to see that the `.nx-tag__text` is the first child and the element passed in as `selectedIcons` is the second child.

Reply:
> > Test that the text is the first child (i.e. it appears before the icon): [597f949](https://github.com/sonatype/sonatype-react-shared-components/commit/597f949381fd36bf9a6681f57984e7127cc12ddc)
> 
> That commit does not succeed in its purpose. Observe that if you change the implementation to put the icon before the text the tests will still pass. You need a test of `NxTag` (not `NxSelectableTag`) in which the `selectedIcons` prop is used and the `.children()` are inspectd to see that the `.nx-tag__text` is the first child and the element passed in as `selectedIcons` is the second child.

Improved: https://github.com/sonatype/sonatype-react-shared-components/pull/290/commits/bab046244ed8dc97a622109acfec9d7aa8948d9a